source/docs/software/wpilib-tools/glass/mech2d-widget.rst

```diff
@@ -0,0 +1,92 @@
+        :language: java
+        :lines: 46-49
+        :linenos:
+        :lineno-start: 46
+
+  .. group-tab:: C++
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/ff52f207ccb685748fb71e1c1726ee98130e4c81/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp
+        :language: cpp
+        :lines: 61-66
+        :linenos:
+        :lineno-start: 61
+
+Then, publish the ``Mechanism2d`` object to NetworkTables:
+
+.. tabs::
+
+  .. group-tab:: Java
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/ff52f207ccb685748fb71e1c1726ee98130e4c81/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
+        :language: java
+        :lines: 51-52
+        :linenos:
+        :lineno-start: 51
+
+  .. group-tab:: C++
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/ff52f207ccb685748fb71e1c1726ee98130e4c81/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp
+        :language: cpp
+        :lines: 34-35
+        :linenos:
+        :lineno-start: 34
+
+.. note:: The ``Mechanism2d`` instance can also be sent using a lower-level NetworkTables API or using the :ref:`Shuffleboard API <docs/software/wpilib-tools/shuffleboard/getting-started/shuffleboard-displaying-data:Displaying data from your robot>`. In this case, the ``SmartDashboard`` API was used, meaning that the :guilabel:`Mechanism2d` widget will appear under the ``SmartDashboard`` table name.
+
+To manipulate a ligament angle or length, call ``setLength()`` or ``setAngle()`` on the ``MechanismLigament2d`` object. When manipulating ligament length based off of sensor measurements, make sure to add the minimum length to prevent 0-length (and therefore invisible) ligaments. Ligament angles are relative to the parent ligament, and follow math notation - the same as :ref:`Rotation2d <docs/software/advanced-controls/geometry/pose:Rotation>` (counterclockwise-positive). A ligament based on the root with an angle of zero will point right.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/ff52f207ccb685748fb71e1c1726ee98130e4c81/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java
+        :language: java
+        :lines: 55-60
+        :linenos:
+        :lineno-start: 55
+
+  .. group-tab:: C++
+
+     .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/ff52f207ccb685748fb71e1c1726ee98130e4c81/wpilibcExamples/src/main/cpp/examples/Mechanism2d/cpp/Robot.cpp
+        :language: cpp
+        :lines: 37-43
+        :linenos:
+        :lineno-start: 37
+
+Viewing the Mechanism2d widget in Glass is very similar to :ref:`viewing the Field2d widget <docs/software/wpilib-tools/glass/field2d-widget:Viewing the Robot Pose in Glass>`.
```

Comment:
```suggestion
Viewing the Mechanism2d widget in Glass is very similar to :ref:`viewing the Field2d widget <docs/software/wpilib-tools/glass/field2d-widget:Viewing the Robot Pose in Glass>`.
```

I'd still like a picture of this in here. It's fine to duplicate a sentence here from the Glass docs.